R/repository.R

```diff
@@ -1,18 +1,23 @@
 #' Create or open a git repository
 #'
-#' Use [git_init()] to start a new repository or [git_clone()] to download a
+#' @description
+#' Use `git_init()` to start a new repository or `git_clone()` to download a
 #' repository from a remote.
 #'
-#' You may use [git_find()] and [git_open()] to explicitly discover and open
+#' You may use `git_find()` and `git_open()` to explicitly discover and open
 #' existing git repositories, but this is usually not needed because all gert
 #' functions also take a path argument which implicitly opens the repo.
 #'
 #' @export
 #' @rdname repository
 #' @name repository
 #' @family git
-#' @param path directory of the git repository. For `git_init` or `git_clone`
-#' this must be a non-existing or empty directory.
+#' @param path directory of the git repository. For [git_init()] or
+#'   [git_clone()] this must be a non-existing or empty directory.
+#' @return
+#' * `git_find()`: path to the git repository
+#' * `git_init()`, `git_open()`: a git repository object
+#' * `git_info()`: a list
```

Comment:
I added return values and reflecting on them raises a couple of questions.

Would it make sense to split the docs based on what sort of object you get back? I see functions that return a path (or even the list) as more user-facing and those that return a `git_repo_ptr'` as more internal facing. In my various experimental git wrappers, I had a global policy that the user talks about the repo via a path and not some S3 or S4 object (which is definitely useful internally).

Would it make sense for `git_init()` to return the path to the repo? Instead of a `git_repo_ptr`?

Reply:
Yes maybe that makes sense. Can you open a PR to see if that breaks anything? Generally, the user doesn't have to work with `git_repo_ptr` at all, you can pass the path instead.